sound/soc/soc-pcm.c

```diff
@@ -2996,6 +3100,55 @@ static int soc_rtdcom_mmap(struct snd_pcm_substream *substream,
 	return -EINVAL;
 }
 
+/*
+ * for virtual FE dai links, there is no need
+ * to register PCM device. So only allocate memory for
+ * the dummy pcm device and substreams to be used for
+ * codec startup
+ */
+static int setup_virtual_fe(struct snd_soc_pcm_runtime *rtd,
+			    struct snd_pcm *pcm)
+{
+	struct snd_pcm_substream *substream;
+	struct snd_pcm_str *pstr;
+	int stream_dir;
+
+	/* create pcm */
+	pcm = kzalloc(sizeof(*pcm), GFP_KERNEL);
+	if (!pcm)
+		return -ENOMEM;
+
+	/* set up playback/capture substream */
+	if (rtd->dai_link->dpcm_playback)
+		stream_dir = SNDRV_PCM_STREAM_PLAYBACK;
+	else
+		stream_dir = SNDRV_PCM_STREAM_CAPTURE;
+
+	pstr = &pcm->streams[stream_dir];
+
+	/* allocate memory for substream */
+	substream = kzalloc(sizeof(*substream), GFP_KERNEL);
+	if (!substream)
+		return -ENOMEM;
+
+	/* define substream */
+	substream->pcm = pcm;
+	substream->pstr = pstr;
+	substream->number = 0;
+	substream->stream = stream_dir;
+	sprintf(substream->name, "subdevice #%i", 0);
```

Comment:
Is the name exposed by /sysff or /proc ? if so, do we have any naming collisions when we register 2 or more VFE ?

Reply:
hmm I did not check if it is exposed. Let me try to make it unique